src/nft/components/collection/CollectionNfts.tsx

```diff
@@ -1,31 +1,178 @@
+import { CollectionSearch, FilterButton } from 'nft/components/collection'
 import { CollectionAsset } from 'nft/components/collection/CollectionAsset'
 import * as styles from 'nft/components/collection/CollectionNfts.css'
-import { Row } from 'nft/components/Flex'
-import { Center } from 'nft/components/Flex'
+import { SortDropdown } from 'nft/components/common/SortDropdown'
+import { Center, Row } from 'nft/components/Flex'
+import { NonRarityIcon, RarityIcon } from 'nft/components/icons'
 import { bodySmall, buttonTextMedium, header2 } from 'nft/css/common.css'
-import { useCollectionFilters, useFiltersExpanded, useIsMobile } from 'nft/hooks'
+import { vars } from 'nft/css/sprinkles.css'
+import {
+  CollectionFilters,
+  initialCollectionFilterState,
+  SortBy,
+  SortByPointers,
+  Trait,
+  useCollectionFilters,
+  useFiltersExpanded,
+  useIsMobile,
+} from 'nft/hooks'
 import { AssetsFetcher } from 'nft/queries'
-import { UniformHeight, UniformHeights } from 'nft/types'
-import { useEffect, useMemo, useState } from 'react'
+import { DropDownOption, GenieAsset, GenieCollection, UniformHeight, UniformHeights } from 'nft/types'
+import qs from 'query-string'
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react'
 import InfiniteScroll from 'react-infinite-scroll-component'
 import { useInfiniteQuery } from 'react-query'
+import { useLocation } from 'react-router-dom'
+
+const rarityStatusCache = new Map<string, boolean>()
+function getRarityStatus(id: string, assets?: (GenieAsset | undefined)[]) {
+  if (rarityStatusCache.has(id)) {
+    return rarityStatusCache.get(id)
+  }
+  const hasRarity = assets && Array.from(assets).reduce((reducer, asset) => !!(reducer || asset?.rarity), false)
+
+  if (hasRarity) {
+    rarityStatusCache.set(id, hasRarity)
+  }
+
+  return hasRarity
+}
+
+const urlParamsUtils = {
+  removeDefaults: (query: any) => {
+    const clonedQuery: Record<string, any> = { ...query }
+
+    // Leveraging default values & not showing them on URL
+    for (const key in clonedQuery) {
+      const valueInQuery = clonedQuery[key]
+      const initialValue = initialCollectionFilterState[key as keyof typeof initialCollectionFilterState]
+
+      if (JSON.stringify(valueInQuery) === JSON.stringify(initialValue)) {
+        delete clonedQuery[key]
+      }
+    }
+
+    // Doing this one manually due to name mismatch - "all" in url, "buyNow" in state
+    if (clonedQuery['all'] !== initialCollectionFilterState.buyNow) {
+      delete clonedQuery['all']
+    }
+
+    const defaultSortByPointer = SortByPointers[initialCollectionFilterState.sortBy]
+    if (clonedQuery['sort'] === defaultSortByPointer) {
+      delete clonedQuery['sort']
+    }
+
+    return clonedQuery
+  },
+
+  // Making values in our URL more state-friendly
+  buildQuery: (query: any, collectionStats: GenieCollection) => {
+    const clonedQuery: Record<string, any> = { ...query }
+
+    ;['traits', 'markets'].forEach((key) => {
```

Comment:
This can be declared as an array then looped through appropriately without the weird `;`